demos/source/aws_iot_demo_jobs.c

```diff
@@ -0,0 +1,591 @@
+
+    if( success )
+    {
+        IotLogInfo( "New Job: %.*s", jsonValueLength, pJsonValue );
+
+        if( jsonValueLength > JOBS_DEMO_MAX_ID_LENGTH )
+        {
+            success = false;
+            IotLogError( "Job ID is too long." );
+        }
+        else
+        {
+            memcpy( _pJobId, pJsonValue, jsonValueLength );
+            _JobIdLength = jsonValueLength;
+        }
+    }
+    else
+    {
+        IotLogError( "Failed to parse JobID from Notify Next." );
+    }
+
+    /* Get the Job Document */
+    if( success )
+    {
+        success = IotJsonUtils_FindJsonValue( pCallbackInfo->u.callback.pDocument,
+                                              pCallbackInfo->u.callback.documentLength,
+                                              JOB_DOC_KEY,
+                                              JOB_DOC_KEY_LENGTH,
+                                              &pJsonValue,
+                                              &jsonValueLength );
+    }
+
+    if( success )
+    {
+        IotLogInfo( "New Job Doc: %.*s", jsonValueLength, pJsonValue );
+
+        if( jsonValueLength > JOBS_DEMO_MAX_JOB_DOC_LENGTH )
+        {
+            success = false;
+            IotLogError( "Job document is too long." );
+        }
+        else
+        {
+            memcpy( _jobDoc, pJsonValue, jsonValueLength );
+            _jobDocLength = jsonValueLength;
+        }
+    }
+    else
+    {
+        IotLogError( "Failed to parse Job Document from Notify Next." );
+    }
+
+    /* Signal Main to continue with the demo. */
+    IotSemaphore_Post( pWaitSem );
+}
+
+/**
+ * @brief A helper to extract "message" from the job document.
+ * @param[out] msg The location of the message buffer.
+ * @param[out] msgLength the length of the message buffer.
+ * @returns a bool. true if successful
+ */
+bool _getMessage( const char ** str,
+                  size_t * strLen )
+{
+    return IotJsonUtils_FindJsonValue( _jobDoc, _jobDocLength, JOB_MESSAGE_KEY, JOB_MESSAGE_KEY_LENGTH, str, strLen );
+}
+
+/**
+ *@brief A helper to extract "action" from the job document.
+ * @param[out] Action The location of the action buffer.
+ * @param[out] actionLength the length of the action buffer.
+ * @returns a bool. true if successful
+ */
+bool _getAction( const char ** str,
+                 size_t * strLen )
+{
+    return IotJsonUtils_FindJsonValue( _jobDoc, _jobDocLength, JOB_ACTION_KEY, JOB_ACTION_KEY_LENGTH, str, strLen );
+}
+
+/**
+ * @brief A helper to extract "topic" from the job document.
+ * @param[out] msg The location of the topic buffer.
+ * @param[out] msgLength the length of the topic buffer.
+ * @returns a bool. true if successful
+ */
+bool _getTopic( const char ** str,
+                size_t * strLen )
+{
+    return IotJsonUtils_FindJsonValue( _jobDoc, _jobDocLength, JOB_TOPIC_KEY, JOB_TOPIC_KEY_LENGTH, str, strLen );
+}
+
+static bool _executeCommand( const char * command,
```

Comment:
The JSON key is "action".  Just so the terminology is consistent, could you either rename the JSON key or replace all instances of "command" with "action"?